packages/core/src/shader/utils/getTestContext.ts

```diff
@@ -11,13 +11,13 @@ let context = unknownContext;
  * @private
  * @returns {WebGLRenderingContext} a gl context to test with
  */
-export function getTestContext()
+export function getTestContext(): WebGL2RenderingContext
```

Comment:
I believe this function returns `WebGLRenderingContext|WebGL2RenderingContext` not just a WebGL2 context.

This also means that `gl` has the same type instead of any

Reply:
That's actually good question I need to clarify: I think its better if we use WebGL2RenderingContext everywhere. All things that we use are the same as in WebGLRenderingContext.

Reply:
Since it’s annoying to maintain webgl|webgl2 in many places, just create an alias type like RenderingContext that combines both. And we can always drop support later for v1. I think it’s important that types be accurate. Just staying webgl2 isn’t accurate.

Reply:
We already use polyfills in a few places that webgl1 doesnt have.

Reply:
But consider building for environments where you explicit are forbidding webgl2. Like some old device browser. We wouldn’t want to fail if webgl2 is disallowed via env types. I don’t understand the harm in being verbose.

Reply:
Our gl-level is beautiful. I do want it to be more verbose, but checks like "gl as WebGL2RenderingContext" might ruin it. 

I propose a compromise: we can make separate interface `IPixiGLContext` and add our polyfills there. However it has to be separate PR so we actually see all the places it affects. That will also help to document things - people will see what did we hack into context and extra article inside jsdoc for `IPixiContext` will prepare them for it. WebGL2 calls can be put to object inside `asGL2` or something more verbose.

Reply:
I’m cool with an interface to insulate ourselves from the problem, as long as  there’s not a lot to cover.

Reply:
I can make alias for WebGL2 for now, and we investigate what to do in separate PR, is that ok?